### PR TITLE
[IMP] hr_payroll: Move wage_type/schedule_pay on contract

### PR DESCRIPTION
**Pull request by tivisse**
Purpose
=======

Currently the wage_type and the schedule pay are defined on the salary structure type level, preventing users paying different employees on the same structure with different salary types or periodicity, which is a employee choice on certain countries, like Mexico or the USA.

Specification
=============

Move the logic at the contract level, but keep the structure type field definition to be used as default values.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_journal.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def _default_invoice_reference_model(self):
         comodel_name='account.account', check_company=True, copy=False, ondelete='restrict',
         string='Default Account',
         domain="[('deprecated', '=', False),"
-               "('account_type', '=', default_account_type), ('account_type', 'not in', ('asset_receivable', 'liability_payable'))]")
+               "'|', ('account_type', '=', default_account_type), ('account_type', 'not in', ('asset_receivable', 'liability_payable'))]")
```
Comment (gawa-odoo) [inline addons/account/models/account_journal.py:97]: I don't know how ready this PR is, but is it something we really want? Letting a user put an asset type account as the default expense account seems dubious to me.

I suppose the reason you want that is to be able to have all the accounts for general type journal? I don't know if we want the receivables or the payables in it too.
If that's the case, I think we should define a domain explicitly for this case, like this for example
```suggestion
               "('account_type', 'not in', ('asset_receivable', 'liability_payable')), '|',
               ('account_type', '=', default_account_type), ('type', '=', 'general')]")
```

Comment (tivisse) [reply addons/account/models/account_journal.py:97]: Hello. Thanks for the feedback :)

This has been discussed with @william-andre when I noticed that the payroll configuration flow was broken due to the 2 commits pointed in the commit message.

I'm not an accounting expert, so I would like to have William's input about this :)

Have a nice day,

Yannick.

**Review by Bertrand2 — APPROVED** [review]

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

File: addons/account/models/account_journal.py (hunk 1 above)
Comment (gawa-odoo) [reply addons/account/models/account_journal.py:96]: So sproutch my remark? :smiling_face_with_tear:

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**